### DISCUSSION TITLE
The future of the keyword (inout) in Mojo 🔥

### DISCUSSION THREAD

**Discussion by mojo-for-ai** [modular/modular#1463] [Ideas]
According to my simple knowledge in Mojo 🔥, I believe that the keyword (inout) performs the same function as the keyword (mutref). Why don’t we get rid of (inout) in favor of (mutref). I mean (inout) confuses any programmer when reading it, as it has no real meaning except to confuse beginners in  Mojo 🔥 I hope with all my heart that Chris Lattner removes this keyword from Mojo 🔥,I also hope that the keyword (borrowed) will be reconsidered. I think it performs the same job as the keyword (ref). Why don’t we delete (borrowed) in favor of (ref)?

**Comment by Eprahim-taha** [comment]
I agree with you, I think that the keyword (inout) It confuses programmers, at least from my personal experience

**Reply by jayzhan211** [reply]
+1 `mut` is more straightforward than `inout`

**Reply by dderooy** [reply]
Considering python devs don't have a concept of immutability, both keywords are new so it doesn't really matter imo. When I started learning Rust I always thought mutref, &mut etc sounds like 'mutt this', 'mutt that'. 

A frivolous opinion but 'inout' sounds nicer in my mind. 

deltaref sounds nice too

**Reply by ksandvik** [reply]
Yes, from Python point of view any keyword is fine as it has to be learned. IMHO inout sounds better than mut, mut means that it's mutated or not, inout means the value went into the function block and comes back, and might or might not have been mutated.
